.github/workflows/ci.yml

```
@@ -0,0 +1,80 @@
+name: Typescript CI Workflow
+
+on:
+  push:
+    branches:
+      - dev
+  pull_request:
+    branches:
+      - dev
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4.0.0
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4.0.0
+        with:
+          node-version: '${{ vars.NODE_VERSION }}'
+          cache: 'npm'
```

Can we try caching the deps here too?
```suggestion
          cache: 'npm'
          cache-dependency-path: package-lock.json
```